### DISCUSSION TITLE
Is there any documentation regarding the special functions used in tensor ?

### DISCUSSION THREAD

**Discussion by Neel-Shah-29** [boostorg/ublas#167] [Q&A]
https://github.com/boostorg/ublas/wiki/Tensor#special-functions
Here the special functions are not described properly in the tensor documentation provided in the wiki.

**Comment by bassoy** [comment]
As you have already pointed out, https://github.com/boostorg/ublas/wiki/Tensor#special-functions is the only place where the tensor  functions are documented. However, that might be not the right place.

It should have been placed in [this](https://github.com/boostorg/ublas/tree/develop/doc/ublas/tensor) place. If you are interested in contributing in documentation, get in contact with @shikharvashistha and have a look at:

https://github.com/boostorg/ublas/discussions/137
https://github.com/boostorg/ublas/projects/5
etc.

**Reply by Neel-Shah-29** [reply]
Hello @shikharvashistha I am interested in contributing to the documentation of tensor . Can you guide me through the process, Can i add the tensor documentation for special functions to https://github.com/boostorg/ublas/tree/develop/doc/ublas/tensor as suggested above ? I will be happy to contribute, Thanks!

**Reply by shikharvashistha** [reply]
> Hello @shikharvashistha I am interested in contributing to the documentation of tensor . Can you guide me through the process, Can i add the tensor documentation for special functions to https://github.com/boostorg/ublas/tree/develop/doc/ublas/tensor as suggested above ? I will be happy to contribute, Thanks!

To proceed with your contributions to the documentation you must learn or get familiar with the following things:

- How Asciidoc works and been implemented in our current documentation.

I would suggest you to build/compile the documentation locally to get a gist of the whole process.

If in case you face any difficulties do let me know.

Best
@shikharvashistha

**Reply by Neel-Shah-29** [reply]
Can you suggest the process of building and compiling the documentation . As a beginner i am not aware of it, just provide some resource to start with the build process

**Reply by shikharvashistha** [reply]
> Can you suggest the process of building and compiling the documentation . As a beginner i am not aware of it, just provide some resource to start with the build process 

There can be two possible ways:

1. Using [boost build](https://github.com/boostorg/build)

2. Using [AsciiDoc](https://docs.asciidoctor.org/asciidoc/latest/)

Regards
@shikharvashistha

**Comment by Neel-Shah-29** [comment]
Hello @shikharvashistha  i have read the ASCII documentation and tried to infer how it works now can you tell which part of documentation from https://github.com/boostorg/ublas/wiki/Tensor needs to be added [here](https://github.com/boostorg/ublas/tree/develop/doc/ublas/tensor). Also state the naming convention of the file that needs to be added

**Reply by shikharvashistha** [reply]
- The proposed changes suggests to add the #special function part wiki to the documentation, so that can be added.

- According to me the name can simply be `special_functions`.

Do let me know if you still face any issues.

Regards
@shikharvashistha

**Reply by Neel-Shah-29** [reply]
@shikharvashistha I have added a pull request for the documentation of special functions kindly have a look at it . Thanks!

**Comment by Neel-Shah-29** [comment]
Hello @shikharvashistha can you please guide me through how to link the file with `special_functions.adoc` as mentioned in this comment https://github.com/boostorg/ublas/pull/170#issuecomment-1049809013 . 
Do i need to add `include :: ublas/tensor/special_functions.adoc` path in this file [here](https://github.com/boostorg/ublas/blob/develop/doc/ublas.adoc)

**Reply by shikharvashistha** [reply]
Hi, I'll suggest to read the [Asciidoctor](https://docs.asciidoctor.org/)  documentation and clear your understanding on the basics.
